### PR TITLE
ASRC: add IPC4 support

### PR DESCRIPTION
**Pull request by softwarecki**
Add support for IPC4 in ASRC module.

Changed buffers allocation method. Each buffer is allocated separately instead of one big blok, exceeding the maximum block size for the memory allocation. This allows to use a ASRC module with more than 2 channels.

### CODE REVIEW

File: src/audio/asrc/asrc.c [hunk 1]
```diff
@@ -367,13 +367,84 @@ static struct comp_dev *asrc_new(const struct comp_driver *drv,
 }
 #endif
 
+static int asrc_initialise_buffers(struct asrc_farrow *src_obj)
```
Comment (lyakh) [inline src/audio/asrc/asrc.c:370]: I see that it was like this before and I might be guilty myself too for letting some of these slip through (or thru :-) ) too, but I think we're trying to use the US spelling throughout the sources, so we could as well use this opportunity to make this `initialize`

File: src/audio/asrc/asrc.c [hunk 2]
```diff
@@ -50,16 +55,23 @@ static const struct comp_driver comp_asrc;
 
 /* c8ec72f6-8526-4faf-9d39-a23d0b541de2 */
 DECLARE_SOF_RT_UUID("asrc", asrc_uuid, 0xc8ec72f6, 0x8526, 0x4faf,
-		 0x9d, 0x39, 0xa2, 0x3d, 0x0b, 0x54, 0x1d, 0xe2);
+		    0x9d, 0x39, 0xa2, 0x3d, 0x0b, 0x54, 0x1d, 0xe2);
```
Comment (lgirdwood) [inline src/audio/asrc/asrc.c:58]: @softwarecki sorry - change from the previous comment, we need to keep both UUIDs until the switchover at MTL. This is to ease the topology usage for distros using Intel platforms.

**Review by lgirdwood — COMMENTED** [review]
LGTM, but just need both UUIDs for the moment.

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@softwarecki can you check CI, there seems to be a lot of errors. Thanks.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by softwarecki** [comment]
In my opinion its ready

**Comment by lgirdwood** [comment]
> In my opinion its ready

Yep, CI is good now. Known issue reported.

**Pull request merged**